src/swarm-engine/Swarm/Game/Step/Const.hs

```diff
@@ -1672,8 +1672,9 @@ execConst runChildProg c vs s k = do
 
   -- The code for grab and harvest is almost identical, hence factored
   -- out here.
-  doGrab :: (HasRobotStepState sig m, Has Effect.Time sig m) => GrabbingCmd -> m Entity
-  doGrab cmd = do
+  -- Optionally defer removal from the world, for the case of the Swap command.
+  doGrab :: (HasRobotStepState sig m, Has Effect.Time sig m) => GrabbingCmd -> Bool -> m Entity
+  doGrab cmd deferRemoval = do
```

Comment:
Can we make a simple enumeration type for use as the second argument to `doGrab` instead of using `Bool`?  Looking at the call sites of `doGrab` it is impossible to know what the `True` or `False` arguments mean.  Even when one does know what the argument is about it is hard to remember which is `True` and which is `False`. It would be much more perspicuous as `doGrab Swap' DeferRemoval` and `doGrab Grab' RemoveNow`, or something similar.

Reply:
Yeah, you can say [Boolean blindness](https://github.com/quchen/articles/blob/master/algebraic-blindness.md#boolean-blindness) and I'll get what you mean :)